cli_042_greenland_ice/contents/src/__init__.py

```diff
@@ -183,80 +299,106 @@ def insertIfNew(newUID, newValues, existing_ids, new_data):
+            filename: filename for source data (string)
+            existing_ids: list of date IDs that we already have in our Carto table (list of strings)
+    RETURN  num_new: number of rows of new data sent to Carto table (integer)
+    '''
     num_new = 0
-
+    # get the contents from source url as a list of lines
     res_rows = tryRetrieveData(SOURCE_URL, filename, TIMEOUT, ENCODING)
+    # create an empty dictionary to store new data (data that's not already in our Carto table)
     new_data = {}
+    # go through each line of content retrieved from source
     for row in res_rows:
+        # get dates, mass, uncertainty index by processing lines that don't start with "HDR"
         if not (row.startswith("HDR")):
+            # split line by space to get dates, mass and uncertainty index as separate elements
             row = row.split()
+            # if length of contents in row matches the length of CARTO_SCHEMA
             if len(row)==len(CARTO_SCHEMA):
                 logging.debug("Processing row: {}".format(row))
-
-                date = decimalToDatetime(row[DATE_INDEX])
-                MASS_INDEX = 1
-                UNCERTAINTY_INDEX = 2
-                values = [date, row[MASS_INDEX], row[UNCERTAINTY_INDEX]]
-
+                # get date by accessing the first element in the list of row
+                date = decimalToDatetime(row[0])
+                # store date, mass index (row[1]) and uncertainty index (row[2]) into a list
+                values = [date, row[1], row[2]]
+                # For new date, check whether this is already in our table. 
+                # If not, add it to the queue for processing
                 new_data = insertIfNew(date, values, existing_ids, new_data)
             else:
                 logging.debug("Skipping row: {}".format(row))
 
+    # if we have found new date to process
     if len(new_data):
         num_new += len(new_data)
+        # create a list of new data
         new_data = list(new_data.values())
+        # insert new data into the carto table
         cartosql.blockInsertRows(CARTO_TABLE, CARTO_SCHEMA.keys(), CARTO_SCHEMA.values(), new_data)
 
     return(num_new)
 
-def get_most_recent_date(table):
-    r = cartosql.getFields(TIME_FIELD, table, f='csv', post=True)
-    dates = r.text.split('\r\n')[1:-1]
-    dates.sort()
-    most_recent_date = datetime.datetime.strptime(dates[-1], '%Y-%m-%d %H:%M:%S')
-    return most_recent_date
+# ATTENTION AMELIA ATTENTION !!!
+# Is this function a candidate for FUNCTIONS FOR CARTO DATASETS
+def updateResourceWatch(num_new):
+    '''
+    This function should update Resource Watch to reflect the new data.
+    This may include updating the 'last update date' and updating any dates on layers
+    '''
+    # If there are new entries in the Carto table
+    if num_new>0:
+        # Update dataset's last update date on Resource Watch
+        most_recent_date = datetime.datetime.utcnow()
```

Comment:
I deleted the function get_most_recent_date(CARTO_TABLE) since I thought datetime.datetime.utcnow() could do the job. Should I put that function back to script?